sky/resources.py

```diff
@@ -1005,10 +999,13 @@ def less_demanding_than(
         if self.disk_tier is not None:
             if other.disk_tier is None:
                 return False
-            if self.disk_tier != other.disk_tier:
-                types = ['low', 'medium', 'high']
-                return types.index(self.disk_tier) < types.index(
-                    other.disk_tier)
+            # Here, BEST tier means the best we can get; for a launched
+            # cluster, the best (and only) tier we can get is the launched
+            # cluster's tier. Therefore, we don't need to check the tier
+            # if it is BEST.
+            if self.disk_tier != resources_utils.DiskTier.BEST:
+                if not self.disk_tier <= other.disk_tier:
```

Comment:
Might be better to add parenthesis for better readability.
```suggestion
                if not (self.disk_tier <= other.disk_tier):
```